coordinator/src/main/java/org/apache/uniffle/coordinator/ServerNode.java

```diff
@@ -110,8 +109,7 @@ public ServerNode(
     this.eventNumInFlush = eventNumInFlush;
     this.timestamp = System.currentTimeMillis();
     this.tags = tags;
-    this.isHealthy = isHealthy;
-    this.status = status;
+    this.status = isHealthy ? status : ServerStatus.UNHEALTHY;
```

Comment:
Should we remove `isHealthy`?

Reply:
Removing the isHealthy state here requires an RPC interface and many test classes need to be modified. I intended to mention a separate PR to modify this state so that the logic here is not too complicated.Just make sure, if you think it needs to be changed here, I'll change it in this PR